src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,141 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int ZERO_HASH = 0;
+    private Node<K, V>[] table;
+    private Node<K, V> currentNode;
+    private int threshold;
+    private int capacity;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            resize();
+        }
+        currentNode = new Node<>(key, value, null);
+        int hash = getIndexByHash(key);
+        if (key == null) {
+            if (table[hash] == null) {
+                table[ZERO_HASH] = currentNode;
+            } else if (table[hash] != null && table[hash].key == null) {
+                table[ZERO_HASH].value = value;
+                return;
+            } else {
+                Node<K, V> prevNode = table[hash];
+                while (prevNode.next != null) {
+                    if (prevNode.next.key == null) {
+                        prevNode.next.value = value;
+                        return;
+                    }
+                    prevNode = prevNode.next;
+                }
+                prevNode.next = currentNode;
+            }
+        } else if (table[hash] == null) {
+            table[hash] = currentNode;
+        } else {
+            if (table[hash].key == currentNode.key || table[hash].key != null
+                        && table[hash].key.equals(currentNode.key)) {
+                table[getIndexByHash(key)].value = currentNode.value;
+                return;
+            } else {
+                Node<K, V> prevNode = table[hash];
+                while (prevNode.next != null) {
+                    if (prevNode.next.key == currentNode.key || prevNode.next.key != null
+                            && prevNode.next.key.equals(currentNode.key)) {
+                        prevNode.next.value = currentNode.value;
+                        return;
+                    }
+                    prevNode = prevNode.next;
+                }
+                prevNode.next = currentNode;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
```

Comment:
A lot of redundancy here. `if (node == null) {} else {}` is enough. 
Check if the current node is null. If it is, it means that there are no nodes in the current bucket (index) of the HashMap for the given key.

If the current node is null, assign the new node (newNode) to that position in the table (bucket).
Increment the size of the HashMap by one because we are adding a new entry.
Exit the method since we have successfully inserted the new node.
If the current node is not null, it means there is already a node present in the current bucket for the given key. In this case, we need to search for the existing node with the same key.

While there are more nodes in the linked list (the node is not null), do the following:
Check if the key of the current node is equal to the given key using Objects.equals().

If the keys are equal, it means we have found a matching node with the same key.
Update the value of the existing node with the new value (value).
Exit the method since we have updated the value.
If the keys are not equal, it means we haven't found a matching node yet. Check if the current node is the last node in the linked list (node.next == null).

If it is the last node, assign the new node (newNode) to the next reference of the current node.

Increment the size of the HashMap by one because we are adding a new entry.

Exit the method since we have successfully inserted the new node.

If it is not the last node, move to the next node in the linked list by assigning node.next to node and continue the loop to search for a matching node.